### DISCUSSION TITLE
Stuttering on skip track

### DISCUSSION THREAD

**Discussion by G0MJW** [rern/rAudio#415] [Bug Reports]
If I skip a track the first few seconds of the next track stutter. This is on a PI3 B+

**Comment by EarsofTin** [comment]
Maybe a buffering problem.  What is the source of the music files - usb drive, NAS, internet radio?  Long ago I had similar problem with NAS files that went away with a new router.

Just speculating but try boosting the buffer. Player>Options>Buffer-Audio.  That may replace stuttering with added lag while the buffer fills.

**Comment by G0MJW** [comment]
USB stick.

**Reply by G0MJW** [reply]
I increased it, it still happens in the first few seconds of each track if you skip, but not if it plays one to the next. Surprised nobody else sees this. I doubled the buffer twice with no effect to 16k. It is a buffer problem as it is always 2-3 seconds into the track. It also happens if you forward within a track, 2-3 seconds later it stutters and is then OK again.

**Comment by G0MJW** [comment]
Is there a fix for this? I can't believe it is not happening to anyone else. It is after all just a standard PI3 with a standard SD card image and a USB stick with some music on it. It is not as if I am using an unusual configuration or anything like that and yet audio stutters a few seconds into the track, only when I skip a track. I know there are some people who never ever skip tracks but surely I am not the only person who does and yet it seems nobody else has this problem so what on earth can it be with the standard build and standard hardware that makes it different in such a way that I am the only one having this problem?

**Reply by cYCL157** [reply]
i usually let mine just play, but took a few minutes to test it out and I could not get mine to have any issue.  i play all of my music from my nas server and the pi is connected to network via wifi.

problem with USB stick?  does same happen on another?

**Comment by G0MJW** [comment]
That’s a good question. I think it might be the PI getting distracted doing something else as a higher priority when it skips tracks, like looking for album art or because whatever the buffers are set to makes no difference. 

 

I don’t think it is the USB stick as surely the buffers would handle that – I think they are set to 8192. I have tried various SB sticks and also keeping music on the SD, which would be preferred except it needs re-writing around 90GB of files each time there is a new image to load.

 

The interesting thing is this only happens on a skip. If you start a track it doesn’t. Could there be a handover problem? It is always 2-3 seconds in and it happens every time, without fail.

 

Mike

**Comment by G0MJW** [comment]
Yes, it still has a lot of problems. Living in the 3rd world with respect to electricity supply reliability, (Rural Oxfordshire, UK, several drops per month), I find the WiFi issue also a show stopper. I tried to explain that it takes many minutes for the network to recover from a power outage, but raudio always gives up far too soon waiting for the network to come up and has to be manually reset after each event.

**Comment by G0MJW** [comment]
Just to add - I installed the latest image and this is still happening. I suspect it is something to do with the browser or the pi doing something else and unable to keep up. No resolution to date has worked. I can't really believe it is happening to nobody else at all.
